packages/web3-eth/src/utils/watch_transaction_for_confirmations.ts

```diff
@@ -1,42 +1,37 @@
-import { DataFormat, EthExecutionAPI, format, PromiEvent } from 'web3-common';
+import {
+	BlockOutput,
+	DataFormat,
+	EthExecutionAPI,
+	format,
+	PromiEvent,
+	Web3BaseProvider,
+} from 'web3-common';
 import { Web3Context } from 'web3-core';
-import { Bytes, numberToHex } from 'web3-utils';
+import { Bytes, HexString32Bytes, numberToHex } from 'web3-utils';
 
 import {
 	TransactionMissingReceiptOrBlockHashError,
 	TransactionReceiptMissingBlockNumberError,
 } from '../errors';
 import { ReceiptInfo, SendSignedTransactionEvents, SendTransactionEvents } from '../types';
 import { getBlockByNumber } from '../rpc_methods';
+import { NewHeadsSubscription } from '../web3_subscriptions';
 
-export function watchTransactionForConfirmations<
-	PromiEventEventType extends SendTransactionEvents | SendSignedTransactionEvents,
-	ReturnFormat extends DataFormat,
->(
-	web3Context: Web3Context<EthExecutionAPI>,
-	transactionPromiEvent: PromiEvent<ReceiptInfo, PromiEventEventType>,
-	transactionReceipt: ReceiptInfo,
-	transactionHash: Bytes,
-	returnFormat: ReturnFormat,
-) {
-	if (
-		transactionReceipt === undefined ||
-		transactionReceipt === null ||
-		transactionReceipt.blockHash === undefined ||
-		transactionReceipt.blockHash === null
-	)
-		throw new TransactionMissingReceiptOrBlockHashError({
-			receipt: transactionReceipt,
-			blockHash: format({ eth: 'bytes32' }, transactionReceipt.blockHash, returnFormat),
-			transactionHash: format({ eth: 'bytes32' }, transactionHash, returnFormat),
-		});
-
-	if (transactionReceipt.blockNumber === undefined || transactionReceipt.blockNumber === null)
-		throw new TransactionReceiptMissingBlockNumberError({ receipt: transactionReceipt });
-
-	// TODO - Should check: (web3Context.requestManager.provider as Web3BaseProvider).supportsSubscriptions
-	// so a subscription for newBlockHeaders can be made instead of polling
+type PromiEventEventTypeBase = SendTransactionEvents | SendSignedTransactionEvents;
+type ReturnFormatBase = DataFormat;
+type WaitProps = {
+	web3Context: Web3Context<EthExecutionAPI>;
+	transactionReceipt: ReceiptInfo;
+	transactionPromiEvent: PromiEvent<ReceiptInfo, PromiEventEventTypeBase>;
+	returnFormat: ReturnFormatBase;
+};
 
+const waitByPolling = ({
```

Comment:
Nit picky, but these methods should be name `watchByPolling` and `watchBySubscription` as we are watching the transactions for confirmations and emitting for every confirmation, not just waiting till number of confirmations >= `web3Context.transactionConfirmationBlocks`